### PR TITLE
Put recorder\replayer\app to the teaser module

### PR DESCRIPTION
**Pull request by yuryybk**
PR related to [Zenhub issue](https://app.zenhub.com/workspaces/vision-sdk-5ca52e6c027e3e7234ac7512/issues/mapbox/mapbox-vision/2123)
At first step put Recorder\Replayer\App files to separate Teaser.
No changes with code except imports\packages\some files renaming

### CODE REVIEW

**Comment by yuryybk** [comment]
@mapbox/mapbox-vision-core-admins please protect branch ybk_common_teaser_app and add Travis checks for PRs to this branch

File: teaser/androidTest/java/com/example/commonteaser/ExampleInstrumentedTest.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.example.commonteaser
```
Comment (alexander-kulikovskii) [inline teaser/androidTest/java/com/example/commonteaser/ExampleInstrumentedTest.kt:1]: nit: useless file

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+import com.mapbox.vision.utils.VisionLogger
+import kotlinx.android.synthetic.main.activity_ar_map.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class ArMapActivity : AppCompatActivity(), MapboxMap.OnMapClickListener, OnMapReadyCallback {
+
+    companion object {
+        private var TAG = ArMapActivity::class.java.simpleName
```
Comment (alexander-kulikovskii) [inline teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:38]:
```suggestion
        private val TAG = ArMapActivity::class.java.simpleName
```

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    private var currentRoute: DirectionsRoute? = null
+    private var navigationMapRoute: NavigationMapRoute? = null
+    private var locationComponent: LocationComponent? = null
+
+    private val arLocationEngine by lazy {
+        LocationEngineProvider.getBestLocationEngine(this)
+    }
+
+    private val arLocationEngineRequest by lazy {
+        LocationEngineRequest.Builder(0)
```
Comment (alexander-kulikovskii) [inline teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:57]: maybe add name for parameter? or it's from Java?

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+                    currentRoute = response.body()!!.routes()[0]
+
+                    // Draw the route on the map
+                    if (navigationMapRoute != null) {
+                        navigationMapRoute!!.updateRouteVisibilityTo(false)
+                    } else {
+                        navigationMapRoute = NavigationMapRoute(null, mapView, mapboxMap, R.style.NavigationMapRoute)
+                    }
+                    navigationMapRoute!!.addRoute(currentRoute)
```
Comment (alexander-kulikovskii) [inline teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:178]:
```suggestion

if (navigationMapRoute == null) {
   navigationMapRoute =  NavigationMapRoute(null, mapView, mapboxMap, R.style.NavigationMapRoute)
}

navigationMapRoute?.let {
   updateRouteVisibilityTo(false)
   addRoute(currentRoute)
}
```

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArNavigationActivity.kt [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+import kotlinx.android.synthetic.main.activity_ar_navigation.back
+
+class ArNavigationActivity : AppCompatActivity(), RouteListener, ProgressChangeListener,
+    OffRouteListener {
+
+    companion object {
+        private var TAG = ArNavigationActivity::class.java.simpleName
+
+        private const val LOCATION_INTERVAL_DEFAULT = 0L
+        private const val LOCATION_INTERVAL_FAST = 1000L
```
Comment (alexander-kulikovskii) [inline teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArNavigationActivity.kt:47]: time unit? `MS`

File: teaser/src/main/res/values/developer-config.xml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Please put your MAPBOX_ACCESS_TOKEN here -->
+<resources>
+    <string name="mapbox_access_token">pk.eyJ1IjoieXVyaWJrIiwiYSI6ImNrN29scHRtYzBhajYzZ3A5aWdxMWw3Y3UifQ.-4b25FZ4UX22YShloZ9bWQ</string>
```
Comment (alexander-kulikovskii) [inline teaser/src/main/res/values/developer-config.xml:4]: should it be here?

Comment (yuryybk) [reply teaser/src/main/res/values/developer-config.xml:4]: Generally, it is a bad idea to keep a token here, but I don't know the context.
@yunikkk @kiryldz  could you comment please?

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt (hunk 2 above)
Comment (yuryybk) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt]: fixed

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt (hunk 4 above)
Comment (yuryybk) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt]: fixed

File: teaser/src/main/res/values/developer-config.xml (hunk 6 above)
Comment (kiryldz) [reply teaser/src/main/res/values/developer-config.xml:4]: This file should not be under git.
Please take a look at `app/.gitignore` as an example.

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt (hunk 3 above)
Comment (yuryybk) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:57]: yeah, java mabox-core-sdk

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArNavigationActivity.kt (hunk 5 above)
Comment (yuryybk) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArNavigationActivity.kt]: fixed

File: teaser/androidTest/java/com/example/commonteaser/ExampleInstrumentedTest.kt (hunk 1 above)
Comment (yuryybk) [reply teaser/androidTest/java/com/example/commonteaser/ExampleInstrumentedTest.kt]: fixed

File: teaser/src/main/res/values/developer-config.xml (hunk 6 above)
Comment (yuryybk) [reply teaser/src/main/res/values/developer-config.xml:4]: good catch!

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArNavigationActivity.kt [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+import com.mapbox.vision.performance.ModelPerformanceRate
+import com.mapbox.vision.utils.VisionLogger
+import kotlinx.android.synthetic.main.activity_ar_navigation.*
+import kotlinx.android.synthetic.main.activity_ar_navigation.back
+
+class ArNavigationActivity : AppCompatActivity(), RouteListener, ProgressChangeListener,
+    OffRouteListener {
+
+    companion object {
+        private var TAG = ArNavigationActivity::class.java.simpleName
```
Comment (alexander-kulikovskii) [inline teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArNavigationActivity.kt:44]:
```suggestion
        private val TAG = ArNavigationActivity::class.java.simpleName
```
nit

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+    private fun getRoute(origin: Point, destination: Point) {
+        NavigationRoute.builder(this)
+            .accessToken(Mapbox.getAccessToken()!!)
+            .origin(origin)
+            .destination(destination)
+            .build()
+            .getRoute(object : Callback<DirectionsResponse> {
+                override fun onResponse(call: Call<DirectionsResponse>, response: Response<DirectionsResponse>) {
+                    if (response.body() == null || response.body()!!.routes().size < 1) {
```
Comment (alexander-kulikovskii) [inline teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:166]:
```suggestion
                    if (response.body() == null || response.body()!!.routes().size == 0) {
```
nit: Maybe it's more obvious

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+                        return
+                    }
+
+                    currentRoute = response.body()!!.routes()[0]
```
Comment (alexander-kulikovskii) [inline teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:170]: and what about other routes? should we process them?

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt (hunk 8 above)
Comment (alexander-kulikovskii) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:166]: or just `response.body()!!.routes().isEmpty()`

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/recorder/RecordingActivity.kt [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+            VisionManager.stop()
+            VisionManager.destroy()
+            visionManagerWasInit = false
+        }
+    }
+
+    private fun toggleRecording() {
+        when (recording_view.state) {
+            RecordingView.State.NotRecording -> {
+                VisionManager.startRecording("$BASE_SESSION_PATH/${dateFormat.format(Date(System.currentTimeMillis()))}")
```
Comment (alexander-kulikovskii) [inline teaser/src/main/kotlin/com/mapbox/vision/teaser/recorder/RecordingActivity.kt:112]: how about small function for `${dateFormat.format(Date(System.currentTimeMillis()))}` ?

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt (hunk 9 above)
Comment (yuryybk) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:170]: @yunikkk @kiryldz WDYT?

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArNavigationActivity.kt (hunk 7 above)
Comment (yuryybk) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArNavigationActivity.kt]: fixed

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt (hunk 8 above)
Comment (yuryybk) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt]: fixed

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/recorder/RecordingActivity.kt (hunk 10 above)
Comment (yuryybk) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/recorder/RecordingActivity.kt]: fixed

File: teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt (hunk 9 above)
Comment (kiryldz) [reply teaser/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt:170]: Definitely not in this PR.

**Pull request merged**